org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/GenerateToStringHandler.java

```diff
@@ -89,11 +90,13 @@ public static WorkspaceEdit generateToString(GenerateToStringParams params, IPro
 			return null;
 		}
 
-		TextEdit edit = generateToString(type, params.fields, monitor);
+		// If cursor position is not specified, then insert to the last by default.
+		IJavaElement insertBefore = CodeGenerationUtils.findInsertElement(type, params.context.getRange());
+		TextEdit edit = generateToString(type, params.fields, insertBefore, monitor);
 		return (edit == null) ? null : SourceAssistProcessor.convertToWorkspaceEdit(type.getCompilationUnit(), edit);
 	}
 
-	public static TextEdit generateToString(IType type, LspVariableBinding[] fields, IProgressMonitor monitor) {
+	public static TextEdit generateToString(IType type, LspVariableBinding[] fields, IJavaElement insertBefore, IProgressMonitor monitor) {

```

Comment:
I'm not quite sure, but maybe call it `elementPosition` is easier to understand than `insertBefore`? That's the argument name in the signature of `GenerateToStringOperation.createOperation()`

Reply:
I would change the variable name to `insertPosition`.